### ISSUE TITLE
Calculations inside inductive lemmas

### ISSUE THREAD

**Issue by jiplucap** [dafny-lang/dafny#81]
I'm not totally sure but I think that written calc{ <some text>} inside the body of an inductive lemma rises a syntactic error.

**Comment by RustanLeino** [comment]
I tried reproducing this.  Indeed, if I write:

```
inductive predicate P()
{
  calc { }
}
```

I get a syntax error (`invalid UnaryExpression`, it says).  This is correct behavior.  The body of the inductive predicate declaration expects an expression, but `calc { }` is not an expression (even if the body of the `calc` is nonempty).  To be an expression, the `calc` construct must be followed by an expression.  There are several similar expressions in Dafny:

```
MyLemma(); E
assert Q; E
assume Q; E
calc { } E
```

These forms are like a statement followed by an expression.  That is, the syntactic form of these expressions is like `Statement Expression`, but the `Statement` is limited to a small set of statements.  The value of the expression is just `E`, but the preceding lemma call, assert statement, assume statement, or calculation are used to help the proof of well-formedness of `E`.

Could it be that the syntax error you thought you got was due to a `calc { }` that was no followed by an expression?

  Rustan
